templatesProvider.js

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * 
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ * 
+ *******************************************************************************/
+
+'use strict';
+
+const { exec } = require('child_process');
+const os = require('os');
+const fs = require('fs');
+
+
+const CODEWIND_ODO_EXTENSION_BASE_PATH = '/codewind-workspace/.extensions/codewind-odo-extension';
+const MASTER_INDEX_JSON_FILE = CODEWIND_ODO_EXTENSION_BASE_PATH + '/templates/master-index.json';
+const RECONCILED_INDEX_JSON_FILE = CODEWIND_ODO_EXTENSION_BASE_PATH + '/templates/index.json';
+const JSON_FILE_URL = 'file://' + RECONCILED_INDEX_JSON_FILE;
+const ODO_CATALOG_LIST_COMMAND = CODEWIND_ODO_EXTENSION_BASE_PATH + '/bin/odo catalog list components | tail -n+2';
```

Comment:
Seems like the latest odo change the output of `odo catalog list components`
```
[root@codewind-workspacelbxixnzo5l5ot71g-7d9948745c-dx5gs bin]# ./odo catalog list components
Odo Supported OpenShift Components:
NAME       PROJECT       TAGS
nodejs     openshift     10,8,8-RHOAR,latest

OpenShift Components:
NAME        PROJECT       TAGS
dotnet      openshift     2.0,latest
httpd       openshift     2.4,latest
nginx       openshift     1.10,1.12,1.8,latest
nodejs      openshift     6
perl        openshift     5.24,5.26,latest
php         openshift     7.0,7.1,latest
python      openshift     2.7,3.5,3.6,latest
ruby        openshift     2.3,2.4,2.5,latest
wildfly     openshift     10.0,10.1,11.0,12.0,13.0,8.1,9.0,latest
```
So we need to change our scripts to accommodate odo's latest change.

Reply:
Ok will update odo output processing logic to handle this new format